spanner_orm/tests/condition_test.py

```diff
@@ -15,12 +15,132 @@
 """Tests for spanner_orm.condition."""
 
 import logging
+import os
 import unittest
 
+from absl.testing import parameterized
+from google.cloud.spanner_v1.proto import type_pb2
+
 import spanner_orm
+from spanner_orm import condition
+from spanner_orm import error
+from spanner_orm.testlib.spanner_emulator import testlib as spanner_emulator_testlib
+from spanner_orm.tests import models
+
+
+class ConditionTest(
+    spanner_emulator_testlib.TestCase,
+    parameterized.TestCase,
+):
+
+  def setUp(self):
+    super().setUp()
+    self.run_orm_migrations(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'migrations_for_emulator_test',
+        ))
+
+  @parameterized.named_parameters(
+      (
+          'minimal',
+          condition.ArbitraryCondition(
+              'FALSE',
+              segment=condition.Segment.WHERE,
+          ),
+          {},
+          {},
+          'FALSE',
+          (),
+      ),
+      (
+          'full',
+          condition.ArbitraryCondition(
+              '$key = IF($true_param, ${key_param}, $value_1)',
+              dict(
+                  key=models.SmallTestModel.key,
+                  true_param=condition.Param(
```

Comment:
Do you think we may want to try to infer the type from the python type? This looks a bit verbose to me, and I think condensing the number of line `condition.Param` takes up (enabled by allowing the type to be omitted) would help.

Reply:
I was thinking about that but decided to start with it being explicit. Happy to add inference in another PR. (The inference code ended up being a decent hunk of code.)

Reply:
Sounds good! Looking forward to a follow-up :)